### ISSUE TITLE
Extension stuck in Status "Activating"

### ISSUE THREAD

**Issue by michaelaflores** [apexskier/nova-typescript#64]
The extension says it will use a bundled version of TypeScript if none are found in node_modules of the project. I have a JS project with no installed version of TS, and the extension is stuck in "activating" status. I'm not sure if it's supposed to emit a warning or something else here, but even if I add a TS file to the project it doesn't begin operating. I was hoping this would add some basic code completion/intellisense-like features even to a JS project, based on the description.

**To Reproduce**
1. Open a JS-only project
2. Observe the extension status in the sidebar

**Expected behavior**
Installing the extension and opening a JS-only project should still provide some extension functionality.

**Screenshots**
<img width="167" alt="Screen Shot 2020-09-12 at 2 53 18 PM" src="https://user-images.githubusercontent.com/11711706/93003839-b5174a00-f507-11ea-96b1-a82708474599.png">

**Environment (please complete the following information):**
- macOS version: 10.15.6 (19G2021)
- Nova version: 1.0 Build 200961
- Extension version: 1.6.1
- TypeScript version: N/A
- Sidebar information: Status: Activating; Typescript Version (blank)
- NodeJS information:
   - node version: 12.6.1
   - npm version: 6.13.4
   - installation method:  Nova extensions UI

```
```

</details>

**Comment by michaelaflores** [comment]
After awhile of keeping Nova open, perhaps 15 minutes or more, I ended up getting this error pop up: 
<img width="426" alt="Screen Shot 2020-09-12 at 3 21 19 PM" src="https://user-images.githubusercontent.com/11711706/93004363-cb270980-f50b-11ea-8c9c-ee6e85615de6.png">

Not sure where to get additional debug info.

**Comment by apexskier** [comment]
Can you try looking at the nova extension console? (Extensions > Show Extension Console). Look for anything with a Source of "TypeScript" or "TypeScript Language Server"

**Comment by dariye** [comment]
I'm running into the same issue. I've installed node using `asdf`

**Comment by apexskier** [comment]
@dariye Any output in the extension console?

**Comment by michaelaflores** [comment]
@apexskier that helped a lot. It showed a failure to npm install, which I realized was related to my local .npmrc file pointing to an intermediary registry that requires VPN access. Modifying my npmrc to prevent that fixed the issue for me.

**Comment by apexskier** [comment]
I may have found the root cause of this - Nova current has a serious bug where $HOME is not set when it [sets up your login shell](https://library.panic.com/nova/environment-variables/), which means tools like `nvm`, `asdf`, etc probably won't work (confirmed with `nvm`). I've reported it and hope it's fixed fast, since I haven't been able to work around it via extension code.

As a workaround, you can set a $PATH manually to allow nova to find your `node` installation in Preferences > Tools.

**Comment by logancollins** [comment]
@apexskier Nova specifically captures `$HOME` for the environment, but only if the user has environment capture enabled. If you are seeing cases where this is not being captured, you might want to investigate the output of `/usr/bin/env` to ensure that `$HOME` is being included in the output of your login shell.

<img width="691" alt="Screen Shot 2020-09-22 at 4 48 30 PM" src="https://user-images.githubusercontent.com/714246/93948487-73e71d00-fcf3-11ea-95cc-f75a255ef6a6.png">

**Comment by apexskier** [comment]
@logancollins 

Given this as my `~/.bash_profile`
```sh
export NOVA_TEST_1=$HOME/testing
```

`/usr/bin/env | sort` produces this in a terminal (omitting a few Terminal.app specific things):
```env
HOME=/Users/cameronlittle
NOVA_TEST_1=/Users/cameronlittle/testing
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/Wireshark.app/Contents/MacOS
PWD=/Users/cameronlittle
SHELL=/bin/bash
SHLVL=1
TMPDIR=/var/folders/y3/fwtgpzrx2vd0cxbt3xv8ly100000gn/T/
USER=cameronlittle
_=/usr/bin/env
```

And Nova is reading this:

<img width="643" alt="Screen Shot 2020-09-23 at 08 36 39" src="https://user-images.githubusercontent.com/329222/93975708-2791f080-fd78-11ea-8df8-327bfc52f44f.png">

**Comment by apexskier** [comment]
@logancollins I'm also still seeing extensions run in a read-only filesystem like I was during beta, which I don't _think_ would be related (even after a full reinstall, wiping application support and everything)

**Comment by apexskier** [comment]
Just tried this on my work computer (totally fresh Nova install) and I have the same experience with env variables and read-only file system.

**Comment by logancollins** [comment]
That's so weird. I'm going to dig deeper today and see what might be causing `$HOME` to be ignored in resolution of other variables. Nova is pretty much sending them direct from the output of the `env` command. I'll report back if I find anything!

As for the read-only filesystem: I've been unable to reproduce this here. LSP extensions (including yours) has readwrite access, and I can't immediately see any reason that would change other than things like macOS turning on App Translocation protection or something similar. But I'll keep investigating. We're still tracking your bug from the beta internally.

**Comment by logancollins** [comment]
I've been able to make some progress, I think. I believe this might be specific to Bash? Zsh doesn't appear to have the same behavior (`$HOME` resolves correctly).

It appears that `$HOME` is set by Zsh automatically, but not by Bash. Nova can definitely set this environment variable manually when resolving the user's shell environment to ensure it exists. I will file a ticket for the 1.2 release.

**Comment by apexskier** [comment]
I actually just noticed I have some handling in my run.sh file, which might have been related to this HOME missing from PATH issue.

https://github.com/apexskier/nova-typescript/blob/a2a56be1823b1815eff77d19e8cee466ac1ad958/typescript.novaextension/run.sh#L14-L16

**Comment by apexskier** [comment]
@logancollins I think both issues are resolved in 1.2 (HOME not being set, and the readonly warnings), from my testing.

I'll close this if I can get a confirmation from anyone else here.
